src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinBlock.java

```diff
@@ -0,0 +1,89 @@
+package com.gtnewhorizons.neid.mixins.early.minecraft;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+/**
+ * The mixins within here exist to override the way vanilla Minecraft handles harvest levels for blocks. Vanilla by
+ * default creates two arrays, with a length of 16, with the index being the metadata. These arrays get populated with
+ * default values to exist for every possible metadata value, regardless of if a block even has multiple metadata types.
+ * To use this same system with 16-bit metadata values, means taking several GBs of RAM for just that storage on the
+ * blocks. This instead creates a default value(the same defaults as normal vanilla, just not populated for every
+ * metadata value) and then creates a hashmap for lookups which gets populated as needed via setHarvestLevel calls. If a
+ * value is not present for the requested metadata in the map, then the default will be returned instead.
+ */
+@Mixin(Block.class)
+public class MixinBlock {
+
+    private String defaultHarvestTool = null;
```

Comment:
The mixin added fields needs to be annotated with @ Unique and the names prefixed with neid$